lib/maths/unittest/CXMeansOnline1dTest.cc

```diff
@@ -699,7 +699,71 @@ void CXMeansOnline1dTest::testAdaption() {
     // Specifically, the data set starts with one cluster then
     // a new cluster appears and subsequently disappears.
 
-    // TODO
+    test::CRandomNumbers rng;
+
+    TDoubleVec mode1;
+    rng.generateNormalSamples(3000.0, 10000.0, 100u, mode1);
+    TDoubleVec mode2a;
+    rng.generateNormalSamples(3000.0, 10000.0, 100u, mode2a);
+    TDoubleVec mode2b;
+    rng.generateNormalSamples(4000.0, 90000.0, 100u, mode2b);
+    TDoubleVec mode3;
+    rng.generateNormalSamples(3000.0, 10000.0, 1000u, mode3);
```

Comment:
Would it make sense to use different values for `mode3`? This tests the third case that the data shifts to a new level. You may need to add slightly more data in this case before the mode for N(3000.0, 10000.0) is removed: i.e. would be 20% of data after just 1000 new points.